xdsl/dialects/vector.py

```diff
@@ -182,7 +182,9 @@ def verify_(self):
 
         if memref_element_type != self.value_to_store.typ.element_type:
             raise VerifyException(
-                "MemRef element type should match the stored vector type.")
+                "MemRef element type should match the stored vector type. Obtained types were "
```

Comment:
Why not string comprehension here? I don't think we use much `str()` elsewhere in the code

Reply:
I used `str()` to print types nicely (like `!i32`) instead of `IntegerType(parameters=[IntAttr(data=32), SignednessAttr(data=<Signedness.SIGNLESS: 0>)], width=IntAttr(data=32), signedness=SignednessAttr(data=<Signedness.SIGNLESS: 0>))`. 

Do you think I should use something else for the same effect?

Reply:
#395 Filed an issue, I think this can be cleaned up later in a consistent way